Cython/Utility/Buffer.c

```diff
@@ -611,6 +624,153 @@ __pyx_buffmt_parse_array(__Pyx_BufFmt_Context* ctx, const char** tsp)
     return 0;
 }
 
+static int __Pyx_BufFmt_AddOpaqueType(__Pyx_BufFmt_Context* ctx, const char *prefix, const char *postfix) {
+
+  __Pyx_BufFmt_StackElem *head = ctx->head;
+  while (1) {
+    const __Pyx_BufExtendedValidation* ev = head->field->type->extended_validation;
+    if (ev) {
+      while (ev->validation_type) {
+        if (strcmp(ev->prefix, prefix) != 0) continue;
+        int match = 0;
+        switch (ev->validation_type) {
+          case __PYX_BUF_EXTENDED_VALIDATION_NAME:
+            match = strcmp(ev->validation_str, postfix) == 0;
+            break;
+          case __PYX_BUF_EXTENDED_VALIDATION_STARTSWITH:
+            match = strncmp(ev->validation_str, postfix, strlen(ev->validation_str)) == 0;
+            break;
+          case __PYX_BUF_EXTENDED_VALIDATION_REGEX: {
+            PyObject *re = PyImport_ImportModule("re");
+            if (unlikely(!re)) return -1;
+            PyObject *py_result = PyObject_CallMethod(
+              re,
+              "match",
+              "ss",
+              ev->validation_str,
+              postfix
+            );
+            Py_DECREF(re);
+            if (unlikely(!py_result)) return -1;
+            match = PyObject_IsTrue(py_result);
+            Py_DECREF(py_result);
+            if (unlikely(match == -1)) return -1;
+            break;
+          }
+        }
+
+        if (match) {
+          ctx->head = head;
+          goto success_;
+        }
+        ++ev;
+      }
+    }
+    if (head->field != &ctx->root) {
+      // Go up and try again
+      --head;
+      if (ctx->fmt_offset != head->parent_offset) {
+        PyErr_SetString(PyExc_ValueError, "Failed - not at start");
+        return -1;
+      }
+    } else {
+      PyErr_SetString(PyExc_ValueError, "Gone off top of stack");
+      return -1;
+    }
+  }
+
+  success_:
```

Comment:
```suggestion
  success_:;
```
need a `;` to compile numpy due to the definition afterwards (didn't think about it)